lib/proxy.d.mts

```diff
@@ -0,0 +1,15 @@
+interface _MockFunction<F> {
+  (...args: any[]): F;
+  new (...args: any[]): F;
+  then<T>(fn: () => T): Promise<T>;
+  catch(): Promise<void>;
+  finally(fn: () => void): Promise<void>;
+}
```

Comment:
i think we might need to type `__createMock__` specifically. (check failing tests)